api/src/main/java/org/apache/iceberg/DataFile.java

```diff
@@ -99,10 +99,24 @@ public interface DataFile extends ContentFile<DataFile> {
       optional(140, "sort_order_id", IntegerType.get(), "Sort order ID");
   Types.NestedField SPEC_ID = optional(141, "spec_id", IntegerType.get(), "Partition spec ID");
 
+  Types.StructType READABLE_METRICS_VALUE_TYPE =
+      Types.StructType.of(
+          optional(142, "column_size", Types.LongType.get(), "Total size on disk"),
+          optional(143, "value_count", Types.LongType.get(), "Total count, including null and NaN"),
+          optional(144, "null_value_count", Types.LongType.get(), "Null value count"),
+          optional(145, "nan_value_count", Types.LongType.get(), "NaN value count"),
+          optional(146, "lower_bound", Types.StringType.get(), "Lower bound in string form"),
```

Comment:
Is it crazy for us to have a different struct for every value type so we can return structs with the actual bound types? Probably ... just thinking about this a little. We could define them programmatically so it wouldn't be so bad? I guess strings are probably fine since most folks will be reading these values with human eyes and not using them in another system.

Reply:
Sure I guess the counts can be string, though its weird. 

To clarify, are you suggesting:
`Types.StructType.of(readableStruct(LOWER_BOUNDS), readable_struct(UPPER_BOUNDS), ...?)`

or something else?

Reply:
For the other metrics in this struct, did we not have all the counts correctly before? I thought those all came out as longs anyway

Reply:
Initially was done with <metric_type>.<column>, and I changed it from @aokolnychyi 's  comment:  https://github.com/apache/iceberg/pull/5376#issuecomment-1210148871

Not sure if that is what you are referring?

I guess its all subjective, but Anton's way is slightly easier if you want to select columns you want to see metrics for.  This way is easier if you want to select by metric type.

Reply:
I did a lot of work prototyping keeping the original struct (which I think you also bringing up).  That is a ton of work and complexity, in writing schema utils to clone the schema for each struct.  In the end I felt all that complexity is not really worth it, a map is much simpler.

And it is more in line  with what trino has (all the bounds as string).

Reply:
Discussed offline, will look into how hard it is to make into a dynamic struct with the right types, assuming we keep this current hierarchy (readable_metric, col, metric_type)